### ISSUE TITLE
aws_iam_service_linked_role for dynamodb.application-autoscaling.amazonaws.com doesn't handle custom_suffix correctly

### ISSUE THREAD

**Issue by jinnko** [terraform-providers/terraform-provider-aws#4439]
### Terraform Version

```
$ terraform -v          
Terraform v0.11.7
+ provider.aws v1.17.0
```

### Affected Resource(s)

- aws_iam_service_linked_role

### Terraform Configuration Files

```hcl
resource "aws_iam_service_linked_role" "dynamodb-autoscaling" {
  aws_service_name = "dynamodb.application-autoscaling.amazonaws.com"
  custom_suffix    = "DynamoDBTable"
}
```

### Debug Output

```
* aws_iam_service_linked_role.dynamodb-autoscaling: Error creating service-linked role with name dynamodb.application-autoscaling.amazonaws.com: InvalidInput: Custom suffix is not allowed for dynamodb.application-autoscaling.amazonaws.com
	status code: 400, request id: f7467abd-4ed2-11e8-a30e-e390d3733bd5
```

### Expected Behavior

The service linked role should be created correctly and not result in changes on subsequent runs.

### Actual Behavior

1. Creation of the service linked role works when the `custom_suffix` option is omitted, however subsequent plans then try to re-create the resource to remove the implicit `custom_suffix`.

2. With the `custom_suffix` option defined we see the error provided above.

### Steps to Reproduce

1. `terraform plan`
2. `terraform apply`

**Comment by bflad** [comment]
Not all AWS services support `custom_suffix` and the error reported above is from the AWS API: `InvalidInput: Custom suffix is not allowed for dynamodb.application-autoscaling.amazonaws.com`

Have you opened an AWS Support case to request this feature for that service?

**Comment by jinnko** [comment]
I should clarify the situation a bit more.

To get a successfully created resource that doesn't get changed on subsequent plan/apply you need to follow these steps:

1. Create the resource with this code:
```
resource "aws_iam_service_linked_role" "dynamodb-autoscaling" {
  aws_service_name = "dynamodb.application-autoscaling.amazonaws.com"
}
```

2. After the resource is applied, if you run a `terraform plan` you will get a _forced new resource_:

```
-/+ aws_iam_service_linked_role.dynamodb-autoscaling (new resource required)
      id:                                                                                                      "arn:aws:iam::<redacted>:role/aws-service-role/dynamodb.application-autoscaling.amazonaws.com/AWSServiceRoleForApplicationAutoScaling_DynamoDBTable" => <computed> (forces new resource)
      arn:                                                                                                     "arn:aws:iam::<redacted>:role/aws-service-role/dynamodb.application-autoscaling.amazonaws.com/AWSServiceRoleForApplicationAutoScaling_DynamoDBTable" => <computed>
      aws_service_name:                                                                                        "dynamodb.application-autoscaling.amazonaws.com" => "dynamodb.application-autoscaling.amazonaws.com"
      create_date:                                                                                             "" => <computed>
      custom_suffix:                                                                                           "DynamoDBTable" => "" (forces new resource)
      name:                                                                                                    "AWSServiceRoleForApplicationAutoScaling_DynamoDBTable" => <computed>
      path:                                                                                                    "/aws-service-role/dynamodb.application-autoscaling.amazonaws.com/" => <computed>
      unique_id:                                                                                               "AROAJ7Q4AQAXHDZT3TCVU" => <computed>

```

3. On subsequent plan/apply runs it's possible to work around the _forced new resource_ by adding the `custom_suffix` to the resource.  With this no changes are detected in the resource.

```
resource "aws_iam_service_linked_role" "dynamodb-autoscaling" {
  aws_service_name = "dynamodb.application-autoscaling.amazonaws.com"
  custom_suffix    = "DynamoDBTable"
}
```

I therefore don't believe this is a "problem" with upstream, however I do believe it is inconsistent behaviour in that the AWS API is performing magic based on the different `aws_service_name` being passed in.

I haven't tested this with other autoscaling applications, but I have a hunch the behaviour can be predicted based on the fact that the `aws_service_name` includes the `application-autoscaling.amazonaws.com` suffix, and that each service that can be autoscaled will be mapped to it's own `custom_suffix`.

**Comment by jinnko** [comment]
I hadn't considered using `ignore_changes`.  That does indeed workaround the issue well enough for now.  Thanks.

**Comment by fredsig** [comment]
This also happens e when using ecs.application-autoscaling.amazonaws.com. Using ignore_changes as a workaround for now.

**Comment by bflad** [comment]
Pull request submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/8931

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.15.0 of the Terraform AWS Provider, likely later this week. 👍
